tests/environment.py

```diff
@@ -0,0 +1,175 @@
+###############################################################
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+# Copyright (c) 2020 Arnout Vandecappelle (Essensium/Mind)
+# This code is subject to the terms of the BSD+Patent license.
+# See LICENSE file for more details.
+###############################################################
+
+import json
+import os
+import re
+import subprocess
+
+from capi import UCCSocket
+from opts import debug
+import sniffer
+
+
+class AlEntity:
+    '''Abstract representation of a MultiAP device (1905.1 AL Entity).
+
+    Derived classes provide concrete implementations for a specific device (e.g. docker).
+
+    This provides basic information about the entity, e.g. its AL MAC address. How this information
+    is retrieved is implementation-specific.
+
+    It also provides an abstract interface to interact with the entity, e.g. for sending CAPI
```

Comment:
e.g.? What other means of communication would be supported through this abstract interface?

Reply:
Running commands on the device.